examples/micro_speech/main/audio_provider.cc

```diff
@@ -54,15 +54,16 @@ constexpr int32_t history_samples_to_keep =
 constexpr int32_t new_samples_to_get =
     (kFeatureSliceStrideMs * (kAudioSampleFrequency / 1000));
 
+const int32_t kAudioCaptureBufferSize = 80000;
+const int32_t i2s_bytes_to_read = 3200;
+
 namespace {
 int16_t g_audio_output_buffer[kMaxAudioSampleSize];
 bool g_is_audio_initialized = false;
 int16_t g_history_buffer[history_samples_to_keep];
+uint8_t i2s_read_buffer[i2s_bytes_to_read] = {};
```

Comment:
I see, sorry wasn't aware of that. But I propose to declare this buffer as static locally in function where it is only needed, it is neater this way. I push the change (as git commit amend to keep the logging neater).

Reply:
@misterb0407 your approach was correct to be honest.
What I meant was just wrap the declaration under that macro. No need to move it back. :)
I will wait for you to amend the change and force push again.

Reply:
Hi @vikramdattu , ok , I tried your recommendation, but there is compile error, it seems because macro `NO_I2S_SUPPORT` is already define earlier in the code :
`#define NO_I2S_SUPPORT CONFIG_IDF_TARGET_ESP32C2 || \`
`                                       (CONFIG_IDF_TARGET_ESP32C3 \`
`                                       && (ESP_IDF_VERSION < ESP_IDF_VERSION_VAL(4, 4, 0)))`

Do you have any other alternative preference?

Reply:
Just change
```
uint8_t i2s_read_buffer[i2s_bytes_to_read] = {};
```
to 
```
#if !NO_I2S_SUPPORT
uint8_t i2s_read_buffer[i2s_bytes_to_read] = {};
#endif
```
and it should do.

Reply:
Sure @vikramdattu , it is done!